ask-sdk-model-runtime/ask_sdk_model_runtime/serializer.py

```diff
@@ -0,0 +1,395 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights
+# Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the
+# License.
+#
+import sys
+import re
+import json
+import typing
+import decimal
+from datetime import date, datetime
+
+from six import iteritems
+from six import text_type
+from six import integer_types
+from enum import Enum
+
+from typing import cast, Any
+
+from .exceptions import SerializationException
+
+unicode_type = text_type
+
+try:
+    long  # type: ignore
+except NameError:
+    long = int
+
+if typing.TYPE_CHECKING:
+    from typing import TypeVar, Dict, List, Tuple, Union, Any
+    T = TypeVar('T')
+
+
+class Serializer(object):
```

Comment:
Similar to `ApiClient`, let's call this `DefaultSerializer` subclasses from the abstract `Serializer` class.